src/prompts/ddl/ddl-banner.tsx

```diff
@@ -0,0 +1,114 @@
+import { Component, h, createRef, RefObject } from 'preact';
+import { DDLBannerPromptConfig } from '../../core';
+import DeeplinkButton from './deeplink-button';
+
+const styles = {
+    bannerIconStyle: {
+        backgroundRepeat: 'no-repeat',
+        backgroundPosition: 'center',
+        backgroundSize: 'cover'
+    },
+    closeIconStyle: {
+        width: 12,
+        height: 12
+    }
+};
+
+export interface DDLBannerProps {
+    config: DDLBannerPromptConfig;
+    onConfirm: (config: DDLBannerPromptConfig) => void;
+    onCancel: (config: DDLBannerPromptConfig) => void;
+    dimensions: (width: number, height: number) => void;
+}
+
+export class DDLBanner extends Component<DDLBannerProps, never> {
+    private containerRef: RefObject<HTMLDivElement>;
+
+    constructor(props: DDLBannerProps) {
+        super(props);
+
+        this.containerRef = createRef<HTMLDivElement>();
+    }
+
+    componentDidMount() {
+        if (!this.containerRef.current) {
+            return;
+        }
+
+        const { clientWidth, clientHeight } = this.containerRef.current;
+
+        this.props.dimensions(clientWidth, clientHeight);
+    }
+
+    onConfirm = () => {
+        this.props.onConfirm(this.props.config);
+    };
+
+    onCancel = () => {
+        this.props.onCancel(this.props.config);
+    };
+
+    render() {
+        const { config } = this.props;
+
+        const { canonicalLinkUrl, position, labels, colors, imageUrl } = config;
+        const { heading, body, acceptAction } = labels.bannerdl;
+        const { bg, fg, acceptActionBg, acceptActionFg } = colors.bannerdl;
+
+        const classes = `kumulos-prompt kumulos-banner-container kumulos-banner-compact kumulos-prompt-position-${position}`;
+
+        const containerStyle = {
+            backgroundColor: bg,
+            color: fg
+        };
+
+        const actionStyle: h.JSX.CSSProperties = {
+            backgroundColor: acceptActionBg,
+            color: acceptActionFg
+        };
+
+        const bannerIconStyle = {
+            ...styles.bannerIconStyle,
+            backgroundImage: `url(${imageUrl})`
+        };
+
+        return (
+            <div style={containerStyle} class={classes} ref={this.containerRef}>
+                <div class="kumulos-banner-close" onTouchEnd={this.onCancel}>
+                    <svg
```

Comment:
Anything special about this close button / could we use a unicode glyph instead?

(Maybe see in app renderer for an option, think we use a character for close in there)